src/background.ts

```diff
@@ -83,20 +83,25 @@ export class TanagerApi {
     );
   }
 
-  onMessage(message: TanagerMessage) {
+  onMessage(message: TanagerMessage, sender?: browser.runtime.MessageSender) {
     if (message.type === TanagerMessageKey.Generic && message.query) {
       const { variables, query } = message;
       return this.query(query, variables ?? {}, {
         source: TanagerMessageSource.Message,
+        sender,
       });
     }
   }
 
-  onExternalMessage(message: TanagerMessage) {
+  onExternalMessage(
+    message: TanagerMessage,
+    sender?: browser.runtime.MessageSender
+  ) {
     if (message.type === TanagerMessageKey.Generic && message.query) {
```

Comment:
We can rely on the manifest protecting us from someone else trying to get into this, right? Like with other message passing, I'll usually check that the message came from jointoucan, but in this case, jointoucan is the only one that's allowed, so that's not needed?

Reply:
yeah, I am assuming this is the case, we can easily test this out too. If not we can add some options that allow though only certain domains.

Reply:
Tried to send a message from Twitter.com to the extension, could not connect. So yeah the manifest should protect us here.

<img width="615" alt="Screen Shot 2021-01-11 at 9 45 06 AM" src="https://user-images.githubusercontent.com/578259/104218513-b1320080-53f1-11eb-9227-6735ab69b8c7.png">

Reply:
Love it!! Nice!